python_modules/dagster-pipes/dagster_pipes/__init__.py

```diff
@@ -929,12 +930,25 @@ class PipesDefaultMessageWriter(PipesMessageWriter):
     BUFFERED_STDIO_KEY = "buffered_stdio"
     STDERR = "stderr"
     STDOUT = "stdout"
+    INCLUDE_STDIO_IN_MESSAGES_KEY: str = "include_stdio_in_messages"
 
     @contextmanager
     def open(self, params: PipesParams) -> Iterator[PipesMessageWriterChannel]:
         if self.FILE_PATH_KEY in params:
             path = _assert_env_param_type(params, self.FILE_PATH_KEY, str, self.__class__)
-            yield PipesFileMessageWriterChannel(path)
+            channel = PipesFileMessageWriterChannel(path)
+
+            if params.get(self.INCLUDE_STDIO_IN_MESSAGES_KEY):
+                log_writer = PipesDefaultLogWriter()
+                log_writer.message_channel = channel
```

Comment:
That's because `PipesDefaultLogWriter` was going to be a user-facing API, and `message_channel` is not available to users (especially because `PipesDefaultLogWriter` is instantiated even before `open_dagster_pipes` is called). 

That's a pretty unique situation, I don't think we will ever have other log writers depending on internal Pipes components. 

I can see 2 options for going forward: 

1. Keep `PipesDefaultLogWriter` a user-facing API and have the mutable state for `message_channel` one way or another. 
2. Hide `PipesDefaultLogWriter` from our users. Use a bool flag to enable it internally (either on the orchestration side as it's implemented right now or on the external process side).

Reply:
I will go with (2) since that's what we agreed on doing upstack